tensorflow/core/ops/sampling_ops.cc

```diff
@@ -0,0 +1,35 @@
+#include "tensorflow/core/framework/op.h"
+
+namespace tensorflow {
+
+REGISTER_OP("BernoulliSample")
+    .Input("p: float")
+    .Input("a: T")
+    .Input("b: T")
+    .Output("output: T")
+    .Attr("T: numbertype")
+    .Attr("seed: int = 0")
+    .Attr("seed2: int = 0")
+    .Doc(R"doc(
+Sample `a` with probability `p`, or equivalently sample `b` with probability
+(1 - `p`).
+
+p: The probability to sample `a`.
+a: Heads.
+b: Tails.
+output: The sampled results which is either `a` or `b`.
+)doc");
+
+REGISTER_OP("SampleDistributionIndex")
```

Comment:
Ah nevermind; I guess that only works for continuous, exponential family, distributions.